### PR TITLE
refactor(gsm): add `error_category` column to gsm table

### PR DESCRIPTION
**Pull request by Chethan-rao**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR adds `error_category` column to gsm table which is an enum in application. These could be used to take multiple decisions in future but currently these are required for elimination routing in which these categories will serve as bucket name.
Additionally the pr fixes the `last_modified` column not being updated in gsm table during `/update`

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Added new column to GSM table. (`error_category`)

```
curl --location 'http://localhost:8080/gsm' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: test_admin' \
--data '{
  "connector": "stripe",
  "flow" : "flow1",
  "sub_flow": "sub_flow",
  "code": "incorrect_cvc",
  "status": "status1",
  "decision": "retry",
  "message": "xyz",
  "step_up_possible": false,
  "error_category": "issue_with_payment_method"
}'
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/api_models/src/gsm.rs [hunk 1]
```diff
@@ -26,6 +26,67 @@ pub struct GsmCreateRequest {
...
+    Eq,
+    serde::Serialize,
+    serde::Deserialize,
+    strum::EnumString,
+    ToSchema,
+)]
+#[serde(rename_all = "snake_case")]
+#[strum(serialize_all = "snake_case")]
+pub enum ErrorCategory {
+    IssuerDecline,
```
Comment (sai-harsha-vardhan) [inline crates/api_models/src/gsm.rs:50]: We already have `ConnectorErrorType` which is used for different purpose in auto retries flow, and has the following values
`UserError, BusinessError, TechnicalError, UnknownError`

Should we follow the same here? @Chethan-rao 
cc: @jarnura

File: crates/api_models/src/gsm.rs [hunk 2]
```diff
@@ -141,4 +206,8 @@ pub struct GsmResponse {
     pub unified_code: Option<String>,
     /// error message unified across the connectors
     pub unified_message: Option<String>,
+    /// category in which error belongs to
+    pub error_category: Option<String>,
```
Comment (sai-harsha-vardhan) [inline crates/api_models/src/gsm.rs:210]: api-reference should be showing all the possible values in the response too, better to make this an enum

File: crates/diesel_models/src/gsm.rs [hunk 3]
```diff
@@ -39,6 +39,8 @@ pub struct GatewayStatusMap {
     pub step_up_possible: bool,
     pub unified_code: Option<String>,
     pub unified_message: Option<String>,
+    pub error_category: Option<String>,
```
Comment (sai-harsha-vardhan) [inline crates/diesel_models/src/gsm.rs:42]: Please use respective enums here

File: crates/router/src/core/gsm.rs [hunk 4]
```diff
@@ -82,6 +84,8 @@ pub async fn update_gsm_rule(
             step_up_possible,
             unified_code,
             unified_message,
+            error_category: error_category.map(|ec| ec.to_string()),
```
Comment (sai-harsha-vardhan) [inline crates/router/src/core/gsm.rs:87]:
```suggestion
            error_category: error_category.map(ToString::to_string),
```

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**